developer/js/source/lexical-model-compiler/model-pseudoclosure.ts

```diff
@@ -0,0 +1,222 @@
+   * the model itself, as the model will need to refer to the definitions herein.
+   */
+  compilePseudoclosure(): string {
+    let defn: string = '';
+    let PSEUDOCLOSURE = ModelPseudoclosure.COMPILED_NAME;
+    defn += `let ${PSEUDOCLOSURE} = {\n`
+
+    // ----------------------
+    // START - the 'defaults', which are common within the same Developer version.
+    defn += `  defaults: {\n    version: "${this.defaults.version}"`;
+    
+    // Only write out `applyCasing` if and when it is needed.
+    if(this.defaults.applyCasing) {
+      defn += `,\n    applyCasing: ${this.defaults.applyCasing.toString()}`;
+    }
+
+    // Finalizes `defaults`
+    defn += `\n  },`;
+    // END - the 'defaults'
+
+    // ----------------------
+    // START - model-specific definitions (when defined)
+    defn += `  model: {\n`;
+    defn += `    searchTermToKey: ${this.model.searchTermToKey.toString()}`;
+
+    if(this.model.applyCasing) {
+      defn += `,\n    applyCasing: ${this.model.applyCasing.toString()}`;
+    }
+    defn += `\n  }`
+    // END - model-specific definitions
+
+    // ----------------------
+    // START - compiled closures.  Given those definitions, write out the 
+    // pseudoclosure-referencing closures for the needed methods.
+
+    // We should be able to define these closures in-line with the object's
+    // initialization.  Worst-case, we simply move the definitions outside
+    // of the pseudoclosure's init and THEN define/assign these closures to
+    // the object, as references will be available then for sure.
+    if(this.model.applyCasing) {
+      // A major potential issue:  if the user wants to call extra custom functions that they've written.
+      //
+      // `applyCasing` recursion SHOULD be fine if they write `this.applyCasing() and forward all arguments 
+      // appropriately, as it will be known as `applyCasing` on the runtime `this` (`model`) object.
+      //
+      // Similarly, as long as any helper functions are similarly compiled and stored as part of `model`,
+      // they should be accessible too.  The issue would be to actually allow use of extra custom funcs
+      // and include them as part of this object as part of compilation.
+      defn += `,\n  applyCasing: function(caseToApply, text) {
+        return ${PSEUDOCLOSURE}.model.applyCasing(caseToApply, text, ${PSEUDOCLOSURE}.defaults.applyCasing);
+      }`;
+    } else if(this.defaults.applyCasing) {
+      // We can't directly assign from `.defaults`, as initialization-time field reads
+      // are not permitted within JS.  Function references, however, are valid.
+      defn += `,\n  applyCasing: function(caseToApply, text) {
+        return ${PSEUDOCLOSURE}.defaults.applyCasing(caseToApply, text);
+      }`;
+    }
+
+    // if(this.searchTermToKey) {
+    defn += `,\n  searchTermToKey: function(text) {
+      return ${PSEUDOCLOSURE}.model.searchTermToKey(text, ${PSEUDOCLOSURE}.applyCasing);
+    }`;
+    // }
```

Comment:
```suggestion
    defn += `,\n  searchTermToKey: function(text) {
      return ${PSEUDOCLOSURE}.model.searchTermToKey(text, ${PSEUDOCLOSURE}.applyCasing);
    }`;
```

Reply:
I'd rather leave this in; there's a chance we may implement future model templates that have no need to `searchTermToKey` behavior.  It's only commented-out here because all of our current templates _do_ require its use.